src/helpers/insights.ts

```diff
@@ -0,0 +1,59 @@
+import { InsightsClientMethod, InsightsClientPayload } from '../types';
+
+export function readDataAttributes(
+  domElement: HTMLElement
+): {
+  method: InsightsClientMethod;
+  payload: Partial<InsightsClientPayload>;
+} {
+  const method = domElement.getAttribute(
+    'data-insights-method'
+  ) as InsightsClientMethod;
+  const serializedPayload = domElement.getAttribute('data-insights-payload');
+  if (typeof serializedPayload !== 'string') {
```

Comment:
yes, and typeof null === 'object'.
This check is there to make sure what we give to JSON.parse is a string.

Reply:
I'm not sure this is doing what you want because `JSON.parse('')` throws an error. We might be better only relying on the try/catch below.

Reply:
are you're suggesting to remove that check altogether?

Reply:
What gives me the feeling that this check is almost inappropriate is that the error message is (or should be) exactly the same as the try/catch below.

From what I understand this is checking that the payload is not a `boolean` for instance, even though it is JSON parsable.

What we could do is move this check in the try/catch and throw an empty error. Then the top-level catch will throw the correct message error for both.

WDYT?

Reply:
> From what I understand this is checking that the payload is not a boolean for instance, even though it is JSON parsable.
It can only be a string or null, not a boolean.

This checks data-insights-payload is there. It could be `!== null` but since 
 JSON.parse expects a string and reenforced that check. That's all.

Can be rephrased to: `The insights helper expects data-insights-payload attribute` but it's more helpful with hint on the type expected.

If we put everything in the try catch, we can't throw distinct messages for "no attribute found" and "invalid attribute"

Reply:
The message in the try/catch covers both. A missing attribute is in itself an invalid attribute value. I think it's fine removing a code path for this.

We can update the error message in the try/catch to give more information:

```
Unable to parse `data-insights-payload`. This data attribute expects a JSON string.
```

Reply:
reading your comment I think you're looking at the wrong diff.
did you see there `atob`  called now that we serialize to base64?
It's even more important not to have `null` for 2 reasons:
- typescript will raise "null is not assignable to type string"
- atob on chrome will serialize `null`  to `"null"`